File: 프로그래머스/unrated/181922. 수열과 구간 쿼리 4/수열과 구간 쿼리 4.kt
class Solution {
    fun solution(arr: IntArray, queries: Array<IntArray>): IntArray {
        var answer: IntArray = intArrayOf()
        answer = arr
        queries.forEach {
            for (i in it[0]..it[1])
                if (i % it[2] == 0) answer[i] += 1
        }
        return answer
    }
}

File: 프로그래머스/unrated/181922. 수열과 구간 쿼리 4/수열과 구간 쿼리 4.swift
import Foundation

func solution(_ arr:[Int], _ queries:[[Int]]) -> [Int] {
    var newArr = arr
    
    queries.forEach {
        for i in $0[0]...$0[1] {
            if i % $0[2] == 0 {
                newArr[i] += 1
            }
        }
    }
    
    return newArr
}